src/runtime/HalideRuntime.h

```diff
@@ -439,6 +439,12 @@ struct halide_type_t {
         return !(*this == other);
     }
 
+    HALIDE_ALWAYS_INLINE bool operator<(const halide_type_t &other) const {
+        return code < other.code || (code == other.code &&
```

Comment:
The indentation here is misleading re: the paren nesting level. I would make each line be a single clause in the disjunction:
    return code < other.code ||
               (code == other.code && bits < other.bits) ||
               (bits == other.bits && lanes < other.lanes);

Though Andrew's suggestion is also promising to me.

Reply:
Re: the indentation, this was adapted from `Type::operator<`, minus the check for handle info, with similar indentation.

Re: Andrew's suggestion, SGTM, but if we do that we should update `Type::operator<` as well so that the relative sort order between Type and halide_type_t is preserved.